wrpcorrectify/correctify.go

```diff
@@ -0,0 +1,51 @@
+// SPDX-FileCopyrightText: 2022 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package wrpcorrectify
+
+import (
+	"context"
+
+	"github.com/xmidt-org/wrp-go/v3"
+)
+
+// Correctifier applies a series of normalizing options to a WRP message.
+type Correctifier struct {
+	opts []Option
+}
+
+// Option is a functional option for normalizing a WRP message.
+type Option interface {
+	// Correctify applies the option to the given message.
+	Correctify(context.Context, *wrp.Message) error
```

Comment:
Is the intention to allow code outside this package to create options?  If so, some additional bulletproofing might be in order.  However, if no code outside this package should be creating `Option` implementations, then this interface should be sealed:  no exported methods.

Reply:
I was thinking that by making the `Option` something outside code could do, it would make the functionality more extensible & useful for one-off places.

What kind of bulletproofing did you have in mind?